src/init.cpp

```diff
@@ -1183,11 +1211,12 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
             LogPrintf("Loaded %i addresses from peers.dat  %dms\n", node.addrman->size(), GetTimeMillis() - nStart);
         } else {
             // Addrman can be in an inconsistent state after failure, reset it
-            node.addrman = std::make_unique<CAddrMan>(/* deterministic */ false, /* consistency_check_ratio */ check_addrman);
+            node.addrman = std::make_unique<CAddrMan>(&asmap, /* deterministic */ false, /* consistency_check_ratio */ check_addrman);
             LogPrintf("Recreating peers.dat\n");
             adb.Write(*node.addrman);
         }
     }
```

Comment:
47ad0fd2644c4de7a11efa9c9e288f73d6c147e5 nit, why these outer brackets?

Reply:
It limits the scope and makes it obvious that local variables like `asmap` and `adb` are not used outside this code block.